lib/OpenQA/Worker/Job.pm

```diff
@@ -1077,9 +1078,8 @@ sub _upload_asset {
                 local => $local_upload,
                 retries => $retries
             });
-    };
-    log_error($@, channels => \@channels_both, default => 1) if $@;
-
+    }
+    catch ($e) { log_error($e, channels => \@channels_both, default => 1) }
```

Comment:
I just saw that line 1088 is still referring to `$@`. Maybe you can do `$error = $e` here.

Probably that behaviour is not covered by tests, otherwise we should see a failure.

Reply:
Nice catch. I wanted to ask about that. what I dont know is if it is override a previous assignment. like in L1063. Neverthless I change it and waiting for CI results